addOns/soap/src/test/java/org/zaproxy/zap/extension/soap/WSDLFilePassiveScanRuleTestCase.java

```diff
@@ -19,25 +19,39 @@
  */
 package org.zaproxy.zap.extension.soap;
 
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
 
 import java.io.IOException;
 import java.lang.reflect.InvocationTargetException;
+import org.apache.commons.httpclient.URI;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 import org.parosproxy.paros.network.HttpHeader;
 import org.parosproxy.paros.network.HttpMessage;
 
-class WSDLFilePassiveScanRuleTestCase {
+class WSDLFilePassiveScanRuleTestCase extends PassiveScannerTest<WSDLFilePassiveScanRule> {
+
+    @Override
+    protected WSDLFilePassiveScanRule createScanner() {
+        return new WSDLFilePassiveScanRule();
+    }
+
     private HttpMessage wsdlMsg = new HttpMessage();
 
     private static void setContentType(HttpMessage msg, String contentType) {
         msg.getResponseHeader().setHeader(HttpHeader.CONTENT_TYPE, contentType);
     }
 
     @BeforeEach
-    void setUp() {
+    void init() {
+        given(passiveScanData.isPage404(any())).willReturn(true);
```

Comment:
Ummm I had added that for the test at 108, but then I parameterized that test and didn't account for isPage500 here. I'll have to investigate further.

Reply:
Oh it's working for both 404 and 500 because the condition is or'd. I'll add them as separate tests and move the given into the test methods so that it can be more specific.